### PR TITLE
Develop repair parser

### PR DESCRIPTION
**Pull request by HuangFru**
## Why are the changes needed?

*(For example: The Arctic Flink dataStream API has already supported this configuration: 'scan.startup.mode', but it is not available with Flink SQL. We should expose this configuration to users when they are using Flink SQL.)*

## Brief change log

*(For example:)*
  - *Add the 'scan.startup.mode' configuration to the ArcticScanContext in the flink modules*
  - *The flink 1.12 and 1.14 versions are affected.*
  - *Add the documentation in the flink chapter*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/OptimizeCall.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.netease.arctic.ams.server.repair.command;
+
+import com.netease.arctic.ams.server.repair.Context;
+
+public class OptimizeCall implements CallCommand {
+
+  /**
+   * Start, Stop
+   */
+  private String way;
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/OptimizeCall.java]: way -> action

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/UseCall.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.netease.arctic.ams.server.repair.command;
+
+import com.netease.arctic.ams.server.repair.Context;
+
+public class UseCall implements CallCommand {
+
+  private String tableName;
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/UseCall.java]: tableName -> database

**Review by shidayang — CHANGES_REQUESTED** [review]

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/SimpleRegexCommandParser.java [hunk 3]
```diff
@@ -18,22 +18,101 @@
...
+    String[] commandSplit = line.split("\\s+");
+    if (commandSplit.length < 2) {
+      throw new IllegalCommandException("Please check if your command is correct!");
+    }
+
+    switch (commandSplit[0].toUpperCase()) {
+      case ANALYZE:
+        return analyzeCallGenerator.generate(commandSplit[1]);
+      case REPAIR:
+        if (StringUtils.equalsIgnoreCase(commandSplit[2], THROUGH) && !(commandSplit.length < 4)) {
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/SimpleRegexCommandParser.java]: ->  !(commandSplit.length < 4) && StringUtils.equalsIgnoreCase(commandSplit[2], THROUGH)

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/SimpleRegexCommandParser.java [hunk 4]
```diff
@@ -18,22 +18,101 @@
...
+            if (commandSplit.length < 5) {
+              throw new IllegalCommandException("Please check if you enter your SnapshotID!");
+            } else {
+              return repairCallGenerator.generate(commandSplit[1], commandSplit[3], commandSplit[4]);
+            }
+          } else if (StringUtils.equalsIgnoreCase(commandSplit[3], FIND_BACK) ||
+              StringUtils.equalsIgnoreCase(commandSplit[3], SYNC_METADATA)) {
+            return repairCallGenerator.generate(commandSplit[1], commandSplit[3], null);
+          }
+        } else {
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/SimpleRegexCommandParser.java]: Make throw from "else" to "if"

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/SimpleRegexCommandParser.java [hunk 5]
```diff
@@ -18,22 +18,101 @@
...
+          throw new IllegalCommandException("Please check if your command is correct!");
+        }
+      case USE:
+        return useCallGenerator.generate(commandSplit[1]);
+      case OPTIMIZE:
+        if (!(commandSplit.length < 3)) {
+          if (StringUtils.equalsIgnoreCase(commandSplit[1], START) ||
+              StringUtils.equalsIgnoreCase(commandSplit[1], STOP)) {
+            return optimizeCallGenerator.generate(commandSplit[1], commandSplit[2]);
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/SimpleRegexCommandParser.java]: make commandSplit[2] from String to enum

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/SimpleRegexCommandParser.java [hunk 6]
```diff
@@ -18,22 +18,101 @@
...
+          }
+        } else {
+          throw new IllegalCommandException("Please check if your command is correct!");
+        }
+      case REFRESH:
+        if (StringUtils.equalsIgnoreCase(commandSplit[1], FILE_CACHE) && !(commandSplit.length < 3)) {
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/SimpleRegexCommandParser.java]: ->  !(commandSplit.length < 3) && StringUtils.equalsIgnoreCase(commandSplit[1], FILE_CACHE)

**Review by shidayang — CHANGES_REQUESTED** [review]

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/OptimizeCall.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.netease.arctic.ams.server.repair.Context;
+
+public class OptimizeCall implements CallCommand {
+
+  /**
+   * Start, Stop
+   */
+  public enum action {
+    Start, Stop
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/OptimizeCall.java]: START,STOP

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/SimpleRegexCommandParser.java [hunk 8]
```diff
@@ -18,22 +18,92 @@
...
+    }
+
+    switch (commandSplit[0].toUpperCase()) {
+      case ANALYZE:
+        return analyzeCallGenerator.generate(commandSplit[1]);
+      case REPAIR:
+        if (commandSplit.length < 4 || !StringUtils.equalsIgnoreCase(commandSplit[2], THROUGH)) {
+          throw new IllegalCommandException("Please check if your command is correct!");
+        }
+        if (StringUtils.equalsIgnoreCase(commandSplit[3], String.valueOf(RepairCall.way.ROLLBACK))) {
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/SimpleRegexCommandParser.java]: String.valueOf(RepairCall.way.ROLLBACK) -> RepairCall.way.ROLLBACK.name()

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/SimpleRegexCommandParser.java [hunk 9]
```diff
@@ -18,22 +18,92 @@
...
-  public String[] keyword() {
-    return new String[0];
+  public String[] keywords() {
+    return new String[]{
+        ANALYZE,
+        REPAIR,
+        USE,
+        OPTIMIZE,
+        REFRESH,
+        SHOW
```
Comment (shidayang) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/repair/command/SimpleRegexCommandParser.java]: START,STOP,THROUGH..... need to add

**Review by shidayang — APPROVED** [review]
